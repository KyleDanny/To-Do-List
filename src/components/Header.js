const Header = () => (
  <header>
    <h1> The Over<span className="App__title--red">REDUX</span>ing List </h1>
  </header>
);

export default Header;