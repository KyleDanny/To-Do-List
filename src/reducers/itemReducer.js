import { 
  ADD_LIST_ITEM, 
  UPDATE_LIST_ITEM,
  TOGGLE_LIST_ITEM,
  DELETE_LIST_ITEM,
  HIDE_LIST_ITEM,
  SHOW_LIST_ITEM,
} from '../types/index.js'

const itemReducer = (state = [], action) => {
  switch(action.type) {
    case ADD_LIST_ITEM:
      return [...state, action.payload]
    case UPDATE_LIST_ITEM:
      return state.map((item)=> { 
        if (item.id === action.payload.id) {
          return {
            ...item,
            title: action.payload.title,
            description: action.payload.description,
          }
        }
        return item;
      })
    case TOGGLE_LIST_ITEM:
      const referenceList = state.map(item => {
        const idToggled = action.payload.id;
        return item.id === idToggled ? { ...item, done: !item.done } : { ...item };
      });
      return referenceList;
    case DELETE_LIST_ITEM:
      const toDelete = state.find(item => item.id === action.payload.id);
      return state.filter((item, index) => index !== state.indexOf(toDelete));
    case HIDE_LIST_ITEM:
      const hideList = state.map(item => {
        if (item.done && !item.hidden) {
          item.hidden = true;
        }
        return item;
      });
      return hideList;
    case SHOW_LIST_ITEM: 
    const showList = state.map(item => {
      if (item.done && item.hidden) {
        item.hidden = false;
      } 
      return item;
    });
    return showList;
    default: 
      return state;
    }
};

export default itemReducer;