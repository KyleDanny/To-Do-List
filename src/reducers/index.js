import itemReducer from "./itemReducer";
import { combineReducers } from "redux";

const rootReducer = combineReducers({ itemReducer });

export default rootReducer;