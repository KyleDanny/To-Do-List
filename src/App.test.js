import reducers from './reducers';
import Header from './components/Header';

import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

test('reducers', () => {
  let state;
  state = reducers(undefined, {});
  expect(state).toEqual({itemReducer:[]});
});

test('reducer & action ADD', () => {
  let state;
  state = reducers({itemReducer:[{title:'People',description:'Persons',done:false,id:656}]}, {type:'ADD_LIST_ITEM',payload:{title:'Apple',description:'My apples',done:false,id:863}});
  expect(state).toEqual({itemReducer:[{title:'People',description:'Persons',done:false,id:656},{title:'Apple',description:'My apples',done:false,id:863}]});
});

test('reducer & action TOGGLE DONE / NOT DONE', () => {
  let state;
  state = reducers({itemReducer:[{title:'dogs',description:'dogsees',done:true,id:656}]}, {type:'TOGGLE_LIST_ITEM',payload:{title:'dogs',description:'dogsees',done:true,id:656}});
  expect(state).toEqual({itemReducer:[{title:'dogs',description:'dogsees',done:false,id:656}]});
});

test('reducer & action UPDATE', () => {
  let state;
  state = reducers({itemReducer:[{title:'dogs',description:'dogsees',done:false,id:656}]}, {type:'UPDATE_LIST_ITEM',payload:{title:'People',description:'Persons',id:656}});
  expect(state).toEqual({itemReducer:[{title:'People',description:'Persons',done:false,id:656}]});
});

test('reducer & action DELETE', () => {
  let state;
  state = reducers({itemReducer:[{title:'People',description:'Persons',done:true,id:656},{title:'Apple',description:'My apples',done:false,id:863}]}, {type:'DELETE_LIST_ITEM',payload:{title:'People',description:'Persons',done:true,id:656}});
  expect(state).toEqual({itemReducer:[{title:'Apple',description:'My apples',done:false,id:863}]});
});

describe('Components: Header', () => {
  test('renders part of h1', () => {
    render(<Header />);
    const text = screen.getByText(/List/i);
    expect(text).toBeInTheDocument();
  });

  test('Snapshot: renders the header', () => {
    const header = renderer.create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
  });
});
