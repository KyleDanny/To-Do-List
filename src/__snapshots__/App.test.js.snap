// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Header Snapshot: renders the header 1`] = `
<header>
  <h1>
     The Over
    <span
      className="App__title--red"
    >
      REDUX
    </span>
    ing List 
  </h1>
</header>
`;

exports[`Header Snapshot: renders the header 1`] = `
<header>
  <h1>
     The Over
    <span
      className="App__title--red"
    >
      REDUX
    </span>
    ing List 
  </h1>
</header>
`;
